client/src/Actions/UserActions.js

```diff
@@ -2,13 +2,24 @@ import axios from 'axios';
 import { URL } from './index';
 import * as types from './actionTypes';
 
+import { tokenIsValid } from './../Helpers/tokenIsValid'
+
 export const GetUser = ( token ) => dispatch => {
-  // Implement further data-checking.
-  
   dispatch({
     type: types.GETTING_USER_START
   })
-  return axios.get(`${URL}/api/me`, { 
+
+  // Preliminary token validation
+  if (!tokenIsValid(token)) {
+    return dispatch({ 
+      type: types.GETTING_USER_FAILURE, 
+      payload: {
+        error: "The provided token is invalid, I'm afraid! Make sure it's a string of the appropriate length"
+      } 
+    });
+  }
+
+  return axios.get(`${URL}/me`, { 
     headers: {
       'Content-Type': 'application/json',
       'Authorization': `${token}`,
```

Comment:
I think we should add "Bearer" before the token as per[ jwt.io](jwt.io) spec.
```
Bearer <token>
```

Reply:
Some Google-fu has lead me to believe it has to be in the following form: `'Authorization': `Bearer${token}`
Is this correct or do I have to add a space in between `Bearer` and `${token}`? I guess it depends on what you prefer in dealing with the headers.

Reply:
Space between please. 👍